src/KubernetesClient/Informers/Cache/ICache.cs

```diff
@@ -0,0 +1,37 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace k8s.Informers.Cache
+{
+    /// <summary>
+    /// Maintains cache of objects of type <typeparamref name="TResource"/>.
+    /// </summary>
+    /// <typeparam name="TKey">The type of key</typeparam>
+    /// <typeparam name="TResource">The type of resource</typeparam>
+    public interface ICache<TKey, TResource> : IDictionary<TKey, TResource>, IDisposable
+    {
+        /// <summary>
+        /// Current version of cache
+        /// </summary>
+        long Version { get; set; }
+        /// <summary>
```

Comment:
newlines (I'm going to stop commenting on this, but please apply the comment everywhere)